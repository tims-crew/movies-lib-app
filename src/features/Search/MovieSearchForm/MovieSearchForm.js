import React from "react";
import { Input } from "reactstrap";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import actions from "../actions";

import "./MovieSearchForm.scss";

export const MovieSearchForm = props => {

  const onChangeHandler = event => {
    const query = event.target.value;
    props.setSearchText(query);
    if (query.length >= 2) {
      props.history.push(`/search/${query}`);
    } else if (query.length === 0) {
      props.history.push(`/`);
    }
  };

  return (
    <div className="d-flex flex-column w-100">
      <div className="position-relative">
        <Input
          data-testid="search-input"
          type="text"
          name="search"
          placeholder="Search by movie title"
          onChange={onChangeHandler}
          value={props.text}
          className="search-input"
        />
        <i className="fas fa-search search-btn" />
      </div>
    </div>
  );
};

MovieSearchForm.propTypes = {
  text: PropTypes.string,
  setSearchText: PropTypes.func.isRequired
};

const mapStateToProps = ({ search }) => ({ text: search.searchText });

const mapDispatchToProps = dispatch => ({
  setSearchText: text => dispatch(actions.setSearchText(text))
});

const enhauncedMovieSearchForm = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(MovieSearchForm)
);

export default enhauncedMovieSearchForm;
