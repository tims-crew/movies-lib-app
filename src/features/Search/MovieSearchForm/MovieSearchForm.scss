@import "theme";

.search-btn {
  position: absolute;
  color: $gray-500;
  top: 50%;
  right: 10px;
  transform: translateY(-50%);
}
