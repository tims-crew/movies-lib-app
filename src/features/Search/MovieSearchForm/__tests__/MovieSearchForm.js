import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { createMemoryHistory } from "history";

import { MovieSearchForm } from "../MovieSearchForm";

afterEach(cleanup);

describe("MovieSearchForm", () => {
  it("relocates to page with search results for user input", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const setSearchText = jest.fn();
    const { getByTestId } = render(
      <MovieSearchForm setSearchText={setSearchText} history={history} />
    );
    const query = "titanic";
    const input = getByTestId("search-input");
    fireEvent.change(input, { target: { value: query } });
    expect(history.location.pathname).toBe("/search/" + query);
    expect(setSearchText).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MovieSearchForm setSearchText={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
