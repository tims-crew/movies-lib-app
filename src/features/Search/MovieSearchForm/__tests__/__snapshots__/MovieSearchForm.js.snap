// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column w-100"
  >
    <div
      class="position-relative"
    >
      <input
        class="search-input form-control"
        data-testid="search-input"
        name="search"
        placeholder="Search by movie title"
        type="text"
        value=""
      />
      <i
        class="fas fa-search search-btn"
      />
    </div>
  </div>
</DocumentFragment>
`;
